engine/src/variable.cpp

```diff
@@ -1114,6 +1114,24 @@ MCVarref *MCVariable::newvarref(void)
 
 ////////////////////////////////////////////////////////////////////////////////
 
+static bool
+__join_paths(MCSpan<MCNameRef> p_base,
+             MCSpan<MCNameRef> p_extra,
+             MCAutoPointer<MCNameRef[]>& r_out)
+{
+    r_out = new MCNameRef[p_base.size() + p_extra.size()];
+    if (*r_out == nullptr)
+        return false;
+    
+    for(int i = 0; i < p_base.size(); i++)
+        r_out[i] = p_base[i];
+    
+    for(int i = 0; i < p_extra.size(); i++)
+        r_out[i + p_base.size()] = p_extra[i];
+    
+    return true;
+}
```

Comment:
Consider:

```c++
static MCAutoPointer<MCNameRef[]>
__join_paths(MCSpan<MCNameRef> p_base,
             MCSpan<MCNameRef> p_extra)
{
    MCAutoPointer t_result =
        new (nothrow) MCNameRef[p_base.size() + p_extra.size()];
    if (t_result)
    {
        /* Populate t_result */
    }
    return std::move(t_result);
}
```

Then, when you use it, you can write:

```c++
MCAutoPointer<MCNameRef[]> t_full_path =
    __join_paths(MCMakeSpan(m_path, m_length),
                 MCMakeSpan(p_path, p_path_length));
if (!t_full_path)
    return false;
```

This ensures RAII everywhere, and it's the pattern recommended by the C++ Guideline Standard for functions that create an owned value.

BTW I generally recommend `MCMakeSpan()` for creating spans in argument contexts like this.